docs/content/learn/tutorials/component-definition/_index.md

```diff
@@ -0,0 +1,680 @@
+- `components` (optional) - Groups `component` objects which define given component(s) in the component definition. Zero or more `component` objects may be used.
+- `capabilities` (optional) - Defines a group of given capabilities in the component definition. Zero or more `capability` objects may be used. Capabilities are not covered in this tutorial.
+- `back-matter` (optional) – Contains references used within the component definition. Use of `back-matter` is not covered in this tutorial.
+{{% /tab %}}
+{{% tab %}}
+{{< highlight yaml "linenos=table" >}}
+---
+component-definition:
+  uuid: a7ba800c-a432-44cd-9075-0862cd66da6b
+  metadata:
+  import-component-definitions:
+  components:
+  capabilities:
+  back-matter:
+
+{{< /highlight >}}
+
+The root of the OSCAL component definition model is `component-definition`. The `id` property (on line 3) is the document's universally unique identifier (UUID), a unique 128-bit number displayed as a string of hyphenated hexadecimal digits as defined by RFC 4122. OSCAL documents use a version 4 UUID (randomly generated) to uniquely identify the document.
+A `component-definition` contains the following items:
+
+- `metadata` (required) - Provides document metadata for the component definition. This is similar to metadata for other OSCAL models, therefor is not described extensively in this tutorial.
+- `import-component-definitions` (optional) – Allows for grouping of `import-component-definition` items.  Each `import-component-definition` item identifies component definition from other resources from which related information is referenced.  Use of `import-component-definition` is not covered in this tutorial.
+- `components` (optional) - Groups `component` items which define given component(s) in the component definition. Zero or more `component` items may be used.
+- `capabilities` (optional) - Defines a group of given capabilities in the component definition. Zero or more `capability` items may be used. Capabilities are not covered in this tutorial.
+- `back-matter` (optional) – Contains references used within the component definition. Use of `back-matter` is not covered in this tutorial.
+{{% /tab %}}
+{{< /tabs >}}
+
+For simplicity of this tutorial, we will only discuss certain data structures in the sections that follow and identify how they can be used to represent our component definition.
+
+### Defining the Component Definition’s Metadata
+
+Most OSCAL models have a standard metadata syntax, therefore, this is not covered extensively in this tutorial.  There are a few considerations however when authoring metadata for a component definition, such as the `<role>`, `<location>`, and `<party>` elements illustrated in lines 9-18 below.
```

Comment:
This is an XML-specific reference. Can you generalize this to work with any of the XML, JSON, or YAML examples below?